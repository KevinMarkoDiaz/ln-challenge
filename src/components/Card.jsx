import React from 'react'

export const Card = ({
  render = "",
  data = {}
}) => {

  const { imagen, titulo, bajada, autor, imgDescription } = data;

  return (
    <div className={`card--${render}`}>
      <div className={`card__image--${render}`}>
        <img
          src={imagen}
          alt={imgDescription || "imagen"}
        />
      </div>
      <h3 className={`card__title--${render}`}>
        <span>Lead. </span>
        {titulo}
      </h3>
      <p className={`card__description--${render}`}>{bajada}</p>
      <p className={`card__author--${render}`}>{autor}</p>
    
      <span className={`card__separator--${render}`}/>
    </div>
  )
};
