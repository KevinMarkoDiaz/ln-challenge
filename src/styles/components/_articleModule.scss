@mixin module-layout {
    margin: 1rem 0px;
    max-width: 925px;
    height: auto;
    display: grid;
}

@mixin module-footer-text {
    display: flex;
    align-items: center;
    max-height: 2rem;
}

@mixin module-footer-row {
    display: flex;
    flex-direction: row;
}

@mixin module-area-layout {
    .module__layout--left {
        grid-area: a;
    }

    .module__layout--centerup {
        grid-area: b;
    }

    .module__layout--centerdown {
        grid-area: d;
    }

    .module__layout--right {
        grid-area: c;
    }
}

.module__container {

    .module__title {
        font-style: normal;
        font-weight: 700;
        font-size: 24px;
        line-height: 32px;
        font-family: $tipografia-encabezados;
        padding-bottom: 0.5rem;
        border-bottom: 1px solid #E4E4E4;
    }

    .module__layout {
        @include module-layout;
    }

    .module__footer {
        font-style: normal;
        font-weight: 400;
        font-size: 18px;
        line-height: 26px;
        font-family: $tipografia-encabezados;
        padding: 1rem;
        border-bottom: 1px solid #E4E4E4;
        border-top: 1px solid #E4E4E4;
        letter-spacing: -0.2px;

        .module__footer--negrilla {
            display: flex;
            text-transform: uppercase;
            align-items: center;
            max-height: 2rem;
            font-size: 16px;
            line-height: 26px;
            letter-spacing: 0px;
            font-weight: 700;
        }

        .module__footer--point {
            margin: 10px;
            color: #CCCCCC;
            font-size: 1.5rem;           
        }

        .module__footer--text {
            @include module-footer-text;
        }
    }
}

@media screen and (min-width: 426px) and (max-width: 654px) {

    .module__layout {
        @include module-layout;
        grid-template-areas:
            "a c"
            "b d";
        grid-template-rows: 1.25fr 0.75fr;
        grid-template-columns: 1fr 1fr;
        @include module-area-layout;
    }

    .module__footer {
        @include module-footer-row;
    }

    .module__footer--text {
        @include module-footer-text;
    }
}

@media screen and (min-width: 655px) and (max-width: 768px) {

    .module__layout {
        @include module-layout;
        grid-template-areas:
            "a b c"
            "a d c";
        grid-template-rows: 1fr 0.75fr;
        grid-template-columns: 0.95fr 1.15fr 0.9fr;

        @include module-area-layout;
    }

    .module__footer {
        @include module-footer-row;
    }

    .module__footer--text {
        @include module-footer-text;
    }
}

@media screen and (min-width: 769px) {

    .module__layout {
        @include module-layout;
        grid-template-areas:
            "a b c"
            "a d c";
        grid-template-rows: 1fr 0.75fr;
        grid-template-columns: 1.15fr 0.95fr 0.8fr;

        @include module-area-layout;
    }

    .module__footer {
        @include module-footer-row;
    }

    .module__footer--text {
        @include module-footer-text;
    }
}