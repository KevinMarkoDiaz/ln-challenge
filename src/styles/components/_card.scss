@mixin flex-colum {
    display: flex;
    flex-direction: column;
}

@mixin font-title {
    font-family: $tipografia-encabezados;
    font-size: 20px;
    line-height: 28px;
    letter-spacing: -0.2px;
}

/*card regular*/
.card--regular {
    @include flex-colum;
    width: 100%;
    height: 468px;
    align-items: flex-start;

    .card__image--regular {
        width: 100%;
        height: 240px;
        outline: 1px solid #F2F2F2;
        background-color: #BCD6FF;
    }

    .card__title--regular {
        @include font-title;
        padding: 0.5rem 1rem;
        font-weight: 600;

        span {
            color: $tipografia-encabezados-color-azul;
        }
    }

    .card__description--regular {
        font-family: $tipografia-contenido;
        color: $tipografia-contenido-color;
        font-size: 1rem;
        line-height: 24px;
        padding: 0px 1rem;
    }

    .card__author--regular {
        font-family: $tipografia-contenido;
        color: $tipografia-autor-color;
        font-size: 14px;
        font-weight: 700;
        line-height: 16px;
        padding: 0.5rem 1rem;
    }
}

/*card author*/

.card--author {
    @include flex-colum;
    position: relative;
    width: auto;
    height: min-content;
    margin: 20px;
    padding: 1rem 1rem 0px 1rem;
    align-items: center;
    outline: 1px solid #CCCCCC;

    .card__image--author {
        width: 56.57px;
        min-height: 56.57px;
        background-color: #CCCCCC;
        border-radius: 50%;
        order: 2;
        margin: 0.5rem;
    }

    .card__title--author {
        @include font-title;
        ;
        text-align: center;
        max-height: 40%;
        font-weight: 600;
        order: 1;
        padding: 0.5rem;

        span {
            color: $tipografia-encabezados-color-azul;
        }
    }

    .card__description--author {
        display: none;
    }

    .card__author--author {
        font-family: $tipografia-contenido;
        color: $tipografia-autor-color;
        font-size: 16px;
        font-weight: 700;
        line-height: 24px;
        padding-bottom: 1rem;
        order: 3;
    }

    .card__separator--author {
        position: absolute;
        transform: rotate(-180deg);
        background-color: #5A5A5A;
        width: 80px;
        height: 1px;
        left: calc(50% - 80px/2);
        bottom: -1px;
        order: 4;
    }
}

/*card Cinema*/

.card--cinema {
    @include flex-colum;
    width: auto;
    min-height: 574.07px;
    justify-content: space-between;
    align-items: stretch;
    background-color: #1C1C1F;
    color: #F2F2F2;

    .card__image--cinema {
        width: 100%;
        min-height: 450.07px;
        background-color: #BCD6FF;
        outline: 1px solid #F2F2F2;
    }

    .card__title--cinema {
        @include font-title;
        ;
        padding: 0.5rem 1rem;
        font-weight: 400;
    }

    .card__description--cinema {
        display: none;
    }

    .card__author--cinema {
        font-family: Arial, Helvetica, sans-serif;
        font-size: 14px;
        font-weight: 700;
        line-height: 16px;
        padding-top: 0.5rem;
        padding: 0px 1rem 0.5rem 1rem;
    }
}