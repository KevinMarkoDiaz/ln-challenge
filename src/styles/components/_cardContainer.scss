.card__container{
    display: flex;
    flex-direction: column;
}

@media screen and (min-width: 426px) and (max-width: 768px) {

    .card__container{
        display: grid;
        grid-template-columns: repeat(2, 1fr);
        grid-gap: 10px;
        grid-auto-rows: minmax(100px, auto);
    }
}
@media screen and (min-width: 769px) and (max-width: 1024px) {

    .card__container{
        display: grid;
        grid-template-columns: repeat(3, 1fr);
        grid-gap: 10px;
        grid-auto-rows: minmax(100px, auto);
    }
}
@media screen and (min-width: 1024px) {

    .card__container{
        display: grid;
        grid-template-columns: repeat(4, 1fr);
        grid-gap: 10px;
        grid-auto-rows: minmax(100px, auto);
    }
}