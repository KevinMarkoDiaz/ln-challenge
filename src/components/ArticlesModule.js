import React from 'react'
import { products } from '../mock/products'
import { Card } from './Card'

export const ArticlesModule = () => {
    return (
        <div className='module__container'>
            <h4 className='module__title'>Opinión</h4>
            <div className='module__layout'>
                <div className="module__layout--left">
                    <Card render='regular' data={products[0]} />
                </div>
                <div className="module__layout--centerup">
                    <Card render='author' data={products[0]} />

                </div>
                <div className="module__layout--centerdown">

                    <Card render='author' data={products[0]} />
                </div>
                <div className="module__layout--right">
                    <Card render='cinema' data={products[0]} />

                </div>
            </div>
            <div className='module__footer'>            
                <p className='module__footer--negrilla'>editoriales</p>               
                <div className='module__footer--text'><div className='module__footer--point'>·</div>Cuando se dieron cuenta</div>                
                <div className='module__footer--text'><div className='module__footer--point'>·</div>Cuando se dieron cuenta</div>
            </div>
        </div>
    )
}
