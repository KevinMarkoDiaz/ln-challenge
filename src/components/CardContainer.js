import React from 'react'
import { selectRender } from '../helpers/herper'
import { products } from '../mock/products'
import { Card } from './Card'

export const CardContainer = () => {
  return (
    <div className='card__container'>
         {
        products.map((element, i) => <Card key={i} data={element} render={selectRender()} />)
      }
    </div>
  )
}
