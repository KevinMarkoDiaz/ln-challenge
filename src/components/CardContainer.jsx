import React from 'react'
import { selectRender } from '../helpers/helpers';
import { products } from '../mock/products'
import { Card } from './Card'

export const CardContainer = () => {
  return (
    <div className='card__container'>
      {
        products.map((element, i) =>
          <Card
            key={i}
            data={element}
            render={selectRender()}
          />)
      }
    </div>
  )
};
