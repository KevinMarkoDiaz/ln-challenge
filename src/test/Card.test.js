import { render, screen } from '@testing-library/react';
import { Card } from '../components/Card';

describe('components - Cards.js - test', () => {

    test('Should match snapshot showing all the component', () => {
        const { container } = render(<Card />);
        expect(container).toMatchSnapshot();
    });
    test('Should render the card img', () => {
        render(<Card />);
        const imagen = screen.getByRole('img', { name: 'imagen' });
        expect(imagen).toBeInTheDocument();
    });
    test('Should render the card title', () => {
        render(<Card />);
        expect(screen.getByText(/Lead./i)).toBeTruthy();
    })
    test('Should render the correct number of tags', () => {
        const { container } = render(<Card />);
        const titleTag = container.getElementsByTagName('h3');
        expect(titleTag.length).toBe(1);
    })
});