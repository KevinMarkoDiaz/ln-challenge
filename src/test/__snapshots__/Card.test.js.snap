// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - Cards.js - test Should match snapshot showing all the component 1`] = `
<div>
  <div
    class="card--"
  >
    <div
      class="card__image--"
    >
      <img
        alt="imagen"
      />
    </div>
    <h3
      class="card__title--"
    >
      <span>
        Lead. 
      </span>
    </h3>
    <p
      class="card__description--"
    />
    <p
      class="card__author--"
    />
    <span
      class="card__separator--"
    />
  </div>
</div>
`;
